File: apps/api/src/service/GroupService.ts
import GroupRepository from "../repository/GroupRepository";
import { v4 as uuidv4 } from 'uuid';

class GroupService {

    static async getAllGroups() {
        const groups = await GroupRepository.getAllGroups();
        return groups;
    }

    static async getGroupById(group_id: string) {
        const group = await GroupRepository.getGroupById(group_id);
        return group;
    }

    static async createGroup(group_name: string) {
        const group_id = uuidv4();
        const group = await GroupRepository.createGroup(group_id, group_name);
        return group;
    }

    static async updateGroup(group_id: string, group_name: string) {
        const group = await GroupRepository.updateGroup(group_id, group_name);
        return group;
    }

    static async deleteGroup(group_id: string) {
        await GroupRepository.deleteGroup(group_id);
    }

}

export default GroupService;

File: apps/cs-api/Services/GroupService.cs
using cs_api.Models;
using cs_api.Repositories;

namespace cs_api.Services;

public class GroupService : IGroupService
{
    private readonly IGroupRepository _groupRepository;

    public GroupService(IGroupRepository groupRepository)
    {
        _groupRepository = groupRepository;
    }

    public async Task<IEnumerable<Group>> GetAllGroups()
    {
        return await _groupRepository.GetAllGroups();
    }

    public async Task<Group?> GetGroupById(string groupId)
    {
        return await _groupRepository.GetGroupById(groupId);
    }
    public async Task<Group> CreateGroup(string groupName)
    {
        var groupId = Guid.NewGuid().ToString();
        return await _groupRepository.CreateGroup(groupId, groupName);
    }
    public async Task<Group?> UpdateGroup(string groupId, string? groupName)
    {
        return await _groupRepository.UpdateGroup(groupId, groupName);
    }
    public async Task DeleteGroup(string groupId)
    {
        await _groupRepository.DeleteGroup(groupId);
    }
}